index.bs

```diff
@@ -4864,22 +4971,26 @@ a numbered step. If outdented, it (today) is rendered as a bullet in the midst o
-    and any specific policy of the [=[RP]=] regarding unsolicited extensions, i.e., those that were not specified as part of
-    <code>|options|.{{PublicKeyCredentialRequestOptions/extensions}}</code>.
-    In the general case, the meaning of "are as expected" is specific to the [=[RP]=] and which extensions are in use.
-
-    Note:  [=Client platforms=] MAY enact local policy that sets additional [=authenticator extensions=] or
-    [=client extensions=] and thus cause values to appear in the [=authenticator extension outputs=] or
-    [=client extension outputs=] that were not originally specified as part of
-    <code>|options|.{{PublicKeyCredentialRequestOptions/extensions}}</code>. [=[RPS]=] MUST be prepared to handle such
-    situations, whether it be to ignore the unsolicited extensions or reject the assertion. The [=[RP]=] can make this
-    decision based on local policy and the extensions in use.
-
-    Note: Since all extensions are OPTIONAL for both the [=client=] and the [=authenticator=], the [=[RP]=] MUST also be
-    prepared to handle cases where none or not all of the requested extensions were acted upon.
+    <li id='authn-ceremony-verify-extension-outputs'>
+        Verify that the values of the [=client extension outputs=] in |clientExtensionResults| and the [=authenticator extension
+        outputs=] in the <code>[=authdataextensions|extensions=]</code> in |authData| are as expected, considering the [=client
+        extension input=] values that were given in <code>|options|.{{PublicKeyCredentialRequestOptions/extensions}}</code>
+        and any specific policy of the [=[RP]=] regarding unsolicited extensions, i.e., those that were not specified as part of
+        <code>|options|.{{PublicKeyCredentialRequestOptions/extensions}}</code>.
+        In the general case, the meaning of "are as expected" is specific to the [=[RP]=] and which extensions are in use.
+
+        Note:  [=Client platforms=] MAY enact local policy that sets additional [=authenticator extensions=] or
+        [=client extensions=] and thus cause values to appear in the [=authenticator extension outputs=] or
+        [=client extension outputs=] that were not originally specified as part of
+        <code>|options|.{{PublicKeyCredentialRequestOptions/extensions}}</code>. [=[RPS]=] MUST be prepared to handle such
+        situations, whether it be to ignore the unsolicited extensions or reject the assertion. The [=[RP]=] can make this
+        decision based on local policy and the extensions in use.
+
+        Note: Since all extensions are OPTIONAL for both the [=client=] and the [=authenticator=], the [=[RP]=] MUST also be
+        prepared to handle cases where none or not all of the requested extensions were acted upon.
+
+        Note: The [=devicePubKey=] extension has explicit verification procedures, see [[#sctn-device-publickey-extension-verification-get]].
```

Comment:
We might want to clarify the verification procedure of the "main" credential in case of an existing DPK.
Let's assume the multi-device key was compromised (e.g., Multi-device fabric was hacked), but the user has a DPK that is not affected by that compromise (since it is exclusively protected by a distinct HW device).

I would think that the DPK extension including in the assertion can be trusted (DPK not compormised) - even though the main assertion signature would likely not be trusted (multi-device key compromised).

Other opinions?

Reply:
I think that's reasonable, but concretely I'm not sure how we would express that. If the primary credential was known to be compromised, an RP could mark it as "not trusted for new devices" in order not to lock a user out. But is that level of policy suitable for a spec such as this?

Reply:
I don't think we want to require any specific policy here, but we should clarify that - from a security perspective - the device-private-key and the main-credential's-private-key are decoupled and that as a result even if the main-credential's-public-key is considered untrusted, the device-public key could still be trusted.